### PR TITLE
In services, show the aliasSymbol for a type even if it's not accessible in the current scope

### LINKED ISSUE #17322: typeToString for type alias defined in different file does not use the type alias name
**TypeScript Version:**  nightly (2.5.0-dev.20170712)

**Code**

**a.ts**
```ts
import { f } from "./b";
f({ x: 1 });
```

**b.ts**
```ts
export type X = { x: number };
export function f(x: X): void {}
```

Hover over `f`.

**Expected behavior:**

`(alias) f(x: X): void`

**Actual behavior:**

```ts
(alias) f(x: {
    x: number;
}): void
```

### PR DESCRIPTION
**Pull request by andy-ms**
Fixes #17322
This may show an unexpected type due to #17349 -- if there is `type T = string | number` *somewhere* and you happen to also have `string | number` in your file, you will see `T`. But such situations will hopefully be rare.

### CODE REVIEW

File: src/compiler/types.ts [hunk 1]
```diff
@@ -2691,6 +2691,8 @@ namespace ts {
         AddUndefined                    = 1 << 13,  // Add undefined to types of initialized, non-optional parameters
         WriteClassExpressionAsTypeLiteral = 1 << 14, // Write a type literal instead of (Anonymous class)
         InArrayType                     = 1 << 15,  // Writing an array element type
+        UseAliasEvenIfNotAccessible     = 1 << 16,  // For a `type T = ... ` defined in a different file, write `T` instead of its value,
```
Comment (aozgaa) [inline src/compiler/types.ts]: maybe "defined outside the current scope"?

Comment (aozgaa) [inline src/compiler/types.ts]: This seems suspiciously correlated to `UseTypeAliasValue`. I'm not sure what the right way to emphasize the difference is, but it seems like a smell.

Comment (andy-ms) [inline src/compiler/types.ts]: That flag isn't used. (#17779)

**Comment by andy-ms** [comment]
@aozgaa, good to go?

**Comment by aozgaa** [comment]
LGTM

**Pull request merged**